app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -107,15 +114,21 @@ public void updateTasks(String healthCode, List<Task> tasks) {
                     if (task.getFinishedOn() != null) {
                         dbTask.setFinishedOn(task.getFinishedOn());
                         dbTask.setHidesOn(task.getFinishedOn());
+                        isFinishing = true;
+                    }
+                    // Not transactional, so we need to do these one-by-one, not in the batch API which
+                    // returns incomprehensibly low-level errors. This is what can happen:
+                    // both succeed, both fail, np
+                    // task saved, event fails, no further scheduled tasks. So don't allow that
+                    // event saved, task fails, 2nd future task created, existing task still needs to be saved 
+                    //          This is acceptable, so do this and save event first 
+                    if (isFinishing) {
```

Comment:
Can we combine this if block with the previous one? The only way for isFinishing to be true is through the previous if block.